### PR TITLE
refactor: add workspace info in index pattern and asset header and update workspace header

### PR DESCRIPTION
**Pull request by raintygao**
### Description

1. Add workspace info in header of index patterns and assets page when in workspace 
2. Update some page header API usage from render component to button definition
3. fine-tune the padding style for workspace settings page and assets page.
## Screenshot
image
out of workspace 
image
in workspace 
image
out of workspace
image
in workspace

image

image

image

## Testing the changes

Enable useNewHomePage and go to index pattern, assets and workspace pages.

## Changelog

- refactor: add workspace info in index pattern and asset header and update workspace header

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by SuZhou-Joe — DISMISSED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

File: src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx [hunk 1]
```diff
@@ -43,7 +43,9 @@ import { FormattedMessage } from '@osd/i18n/react';
 import { withRouter, RouteComponentProps } from 'react-router-dom';
 import React, { useState, useEffect } from 'react';
 import { i18n } from '@osd/i18n';
-import { useEffectOnce, useMount } from 'react-use';
+import { useEffectOnce } from 'react-use';
+import { useObservable } from 'react-use';
```
Comment (ruanyl) [inline src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx:47]:
```suggestion
import { useEffectOnce, useObservable } from 'react-use';
```

File: src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx [hunk 2]
```diff
@@ -243,13 +246,22 @@ export const IndexPatternTable = ({ canSave, history }: Props) => {
...
-      defaultMessage="Create and manage the index patterns that help you retrieve your data from OpenSearch."
-    />
-  ) as unknown) as string;
-  const pageTitleAndDescription = showActionsInHeader ? (
+  const description = i18n.translate(
+    'indexPatternManagement.indexPatternTable.indexPatternExplanation',
+    currentWorkspace
+      ? {
+          defaultMessage:
+            'Create and manage the index patterns that help you retrieve your data from OpenSearch for {name} workspace. ',
```
Comment (ruanyl) [inline src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx:254]: Nit: seems there is an extra space at the end?

File: src/plugins/saved_objects_management/public/management_section/objects_table/components/header.tsx [hunk 3]
```diff
@@ -36,12 +36,12 @@ import {
   EuiText,
   EuiTextColor,
   EuiButtonEmpty,
-  EuiButton,
 } from '@elastic/eui';
 import { FormattedMessage } from '@osd/i18n/react';
 import { ApplicationStart } from 'src/core/public';
 import { i18n } from '@osd/i18n';
 import { NavigationPublicPluginStart } from '../../../../../navigation/public';
+import { TopNavControlButtonData } from '../../../../../navigation/public';
```
Comment (ruanyl) [inline src/plugins/saved_objects_management/public/management_section/objects_table/components/header.tsx:44]:
```suggestion
import { NavigationPublicPluginStart, TopNavControlButtonData } from '../../../../../navigation/public';
```

**Review by ruanyl — APPROVED** [review]

File: src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx (hunk 2 above)
Comment (raintygao) [reply src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx]: Thanks for catching it. Updated.

**Review by ruanyl — DISMISSED** [review]

**Review by SuZhou-Joe — DISMISSED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Pull request merged**